File: ConexionBD/ConfiguracionBD.cs
using punto_de_venta_C_.Modelo;
using MySqlConnector;
using System;

namespace punto_de_venta_C_.ConexionBD
{
    public class ConfiguracionBD
    {
        public ConfiguracionTienda obtener()
        {
            string sql = "SELECT * FROM configuracion WHERE id = 1";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand st = new MySqlCommand(sql, c))
                {
                    using (MySqlDataReader rs = st.ExecuteReader())
                    {
                        if (rs.Read())
                        {
                            return new ConfiguracionTienda(rs.GetString("nombre_tienda"), rs.GetString("sucursal"), rs.GetString("rfc"));
                        }
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return null;
        }

        public void actualizar(ConfiguracionTienda config)
        {
            string sql = "UPDATE configuracion SET nombre_tienda=@nombre, sucursal=@sucursal, rfc=@rfc WHERE id=1";

            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@nombre", config.getNombre());
                    ps.Parameters.AddWithValue("@sucursal", config.getSucursal());
                    ps.Parameters.AddWithValue("@rfc", config.getRfc());

                    int filasAfectadas = ps.ExecuteNonQuery();
                    Console.WriteLine("Filas actualizadas en BD: " + filasAfectadas);
                }
            }
            catch (Exception e)
            {
                Console.Error.WriteLine("Error al guardar en BD: " + e.Message);
            }
        }
    }
}

File: ConexionBD/ConfiguracionBD.py
from ConexionBD.Conexion import Conexion
from Modelo.ConfiguracionTienda import ConfiguracionTienda

class ConfiguracionBD:
    def obtener(self):
        sql = "SELECT * FROM configuracion WHERE id = 1"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor(dictionary=True)
                cursor.execute(sql)
                row = cursor.fetchone()
                cursor.close()
                if row:
                    config = ConfiguracionTienda()
                    config.setNombre(row["nombre_tienda"])
                    config.setSucursal(row["sucursal"])
                    config.setRfc(row["rfc"])
                    return config
        except Exception as e:
            print(e)
        return None

    def actualizar(self, config):
        sql = "UPDATE configuracion SET nombre_tienda=%s, sucursal=%s, rfc=%s WHERE id=1"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor()
                cursor.execute(sql, (config.getNombre(), config.getSucursal(), config.getRfc()))
                conn.commit()
                filasAfectadas = cursor.rowcount
                cursor.close()
                print(f"Filas actualizadas en BD: {filasAfectadas}")
        except Exception as e:
            print(e)
